kimchi/src/prover.rs

```diff
@@ -340,13 +351,18 @@ where
         //~    As mentioned above, we commit using the evaluations form rather than the coefficients
         //~    form so we can take advantage of the sparsity of the evaluations (i.e., there are many
         //~    0 entries and entries that have less-than-full-size field elemnts.)
-        let witness_poly: [DensePolynomial<G::ScalarField>; COLUMNS] = array::from_fn(|i| {
-            Evaluations::<G::ScalarField, D<G::ScalarField>>::from_vec_and_domain(
-                witness[i].clone(),
-                index.cs.domain.d1,
-            )
-            .interpolate()
-        });
+        let witness_poly: [DensePolynomial<G::ScalarField>; COLUMNS] = (0..COLUMNS)
+            .into_par_iter()
+            .map(|i| {
+                Evaluations::<G::ScalarField, D<G::ScalarField>>::from_vec_and_domain(
```

Comment:
For a follow-up optimisation: we should have a method somewhere that allows use to interpolate directly using a ref on the Vec<F> directly, instead of a ref on `Evaluations` (we do have `interpolate_by_ref` already, but the ref is on `Evaluations`).
```
    /// Interpolate a polynomial from a list of evaluations
    pub fn interpolate_by_ref(&self) -> DensePolynomial<F> {
        DensePolynomial::from_coefficients_vec(self.domain.ifft(&self.evals))
    }
```
gives directly the evals.
We can avoid cloning `witness[i]` L358.

Reply:
I did check that and we could write this function, but it is same or 1% slower because `ifft_in_place` is a bit faster than `ifft` (by reference), so not worth adding more code.